testsupport/wait/awaitility.go

```diff
@@ -796,6 +796,41 @@ func (w *Waiter[T]) WithNameThat(name string, predicates ...assertions.Predicate
 	return returnedObject, err
 }
 
+// WithNameDeleted waits for a single object with the provided name in the namespace of the awaitality to get deleted
+func (w *Waiter[T]) WithNameDeleted(name string) error {
+	w.t.Logf("waiting for object of GVK '%s' with name '%s' in namespace '%s' to be deleted", w.gvk, name, w.await.Namespace)
+	err := wait.Poll(w.await.RetryInterval, w.await.Timeout, func() (done bool, err error) {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(w.gvk)
+		if err := w.await.Client.Get(context.TODO(), client.ObjectKey{Name: name, Namespace: w.await.Namespace}, obj); err != nil {
+			if apierrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
```

Comment:
If there is a non-nil error returned here then it will stop retrying right? Maybe it's better to ignore the error to reduce the chance of test flakiness in cases of connection issues or some other temporary issue? It should still fail with a timeout if the object cannot be deleted due to other issues. WDYT? Also, if we do that we can log the error for debugging purposes.

Reply:
Well.. theoretically I guess you are right. But I don't remember a single case of flaky test because of a connection issue or anything like that. But maybe you've seen it? Returning an error here would make it easier to debug the problem because it would be in the failed test message. It can be hard to look for that error in logs.
And we do it everywhere (or at least in most cases). So I would still keep it as is, so it's consistent at least if not for other reasons.